pkgs/applications/editors/emacs/build-support/generic.nix

```
@@ -82,8 +86,8 @@ stdenv.mkDerivation (finalAttrs: ({
     addEmacsVars "$out"
 
     find $out/share/emacs -type f -name '*.el' -print0 \
-      | xargs -0 -I {} -n 1 -P $NIX_BUILD_CORES sh -c \
-          "emacs --batch --eval '(setq large-file-warning-threshold nil)' -f batch-native-compile {} || true"
+      | xargs --verbose -0 -I {} -n 1 -P $NIX_BUILD_CORES sh -c \
+          "emacs --batch --eval '(setq large-file-warning-threshold nil)' --eval '(setq byte-compile-error-on-warn ${if finalAttrs.turnCompilationWarningToError then "t" else "nil"})' -f batch-native-compile {} || exit ${if finalAttrs.ignoreCompilationError then "0" else "\\$?"}"
```

```suggestion
          "emacs --batch \
            --eval '(setq large-file-warning-threshold nil)' \
            --eval '(setq byte-compile-error-on-warn ${if finalAttrs.turnCompilationWarningToError then "t" else "nil"})' -f batch-native-compile {} \
            || exit ${if finalAttrs.ignoreCompilationError then "0" else "\\$?"}"
```

P.S.: is this call to `exit` fine? As far as I remember, `exit` exits the shell.

---

It's fine.  It is a subshell launched by `xargs` (note the `sh -c`).